vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridMultiSelectionModel.java

```diff
@@ -76,15 +76,15 @@ public AbstractGridMultiSelectionModel(Grid<T> grid) {
 
         selectionColumn
                 .setSelectAllCheckBoxVisibility(isSelectAllCheckboxVisible());
-        grid.getElement().getNode().runWhenAttached(ui -> grid.getElement()
-                .insertChild(0, selectionColumn.getElement()));
+        grid.getElement()
+                .insertChild(0, selectionColumn.getElement());
     }
 
     @Override
     protected void remove() {
         super.remove();
         deselectAll();
-        if (selectionColumn.getElement().getNode().isAttached()) {
+        if (selectionColumn.getElement() != null && getGrid().getElement().getChild(0).equals(selectionColumn.getElement())) {
```

Comment:
Why is this null check needed now? The master version didn't seem to need it.

Reply:
changed code with a new fixing version.